classes/clang.bbclass

```diff
@@ -142,7 +142,9 @@ RECIPESYSROOTFUNCS = ""
 RECIPESYSROOTFUNCS:toolchain-clang = "recipe_sysroot_check_ld_is_lld"
 
 recipe_sysroot_check_ld_is_lld () {
-    if "${@bb.utils.contains('DISTRO_FEATURES', 'ld-is-lld', 'true', 'false', d)}"; then
+    if "${@bb.utils.contains('DISTRO_FEATURES', 'ld-is-lld', 'true', 'false', d)}" && \
+        "${@bb.utils.contains('TARGET_ARCH', 'allarch', 'false', 'true', d)}"
```

Comment:
this is already fixed in master please do a direct backport of 0f31c0b29580825af8700e484b984a8824a31a4f